### PR TITLE
adding scripts isolation

### PR DESCRIPTION
**Pull request by darkyne**
To ease the creation of setup / teardown scripts, a student/scripts directory is automatically mounted in the container. This directory is isolated from the student during the ssh session, allowing the teacher to put scripts files in a safe directory without worying about the student inspecting them.

Documentation was also updated.

### CODE REVIEW

File: base-containers/base/inginious_container_api/utils.py [hunk 1]
```diff
@@ -96,3 +96,14 @@ def ssh_wait(ssh_user, timeout=None):
...
+
+def scripts_isolation(isolate):
+    """
+    Make the script directory isolated or not from the student
+    """
+    if isolate:
+        value = "-rwx"
+    else:
+        value = "+rwx"
+    execute_process(["/usr/bin/bash", "-c", "chmod {} student/scripts".format(value)],
```
Comment (anthonygego) [inline base-containers/base/inginious_container_api/utils.py:108]: Try to use Python code as much as possible here (using the library ``os``).

File: base-containers/base/inginious_container_api/utils.py [hunk 2]
```diff
@@ -96,3 +96,10 @@ def ssh_wait(ssh_user, timeout=None):
     else:
         print("No one connected !")
         return 253  # timeout
+
+def scripts_isolation(isolate):
+    """ Make the script directory isolated or not from the student """
+    if isolate:
+        os.chmod("student/scripts", 000)
```
Comment (anthonygego) [inline base-containers/base/inginious_container_api/utils.py:103]: Did you check this does not fail if the folder does not exist ?

Comment (darkyne) [reply base-containers/base/inginious_container_api/utils.py:103]: Yes, hopefuly I anticipated this one :)
A "student/scripts" directory is always mounted inside the student_container, even if it's empty (and the initial task had no student/scripts directory).
This was just to make it the most general (every student_container thus has a "student/scripts" directory.

Don't hesitate to tell me if you think it's not the right approach

Comment (anthonygego) [reply base-containers/base/inginious_container_api/utils.py:103]: LGTM then. I suggest to let @GuillaumeDerval have a look at his.

**Comment by darkyne** [comment]
Added 2 little lines to resolve #694 so both small features could be added at once.

**Comment by anthonygego** [comment]
Reminder: please change the relative paths to absolute paths to ensure that the scripts will work from everywhere. It should be good for me after that... as no one else seems to care about this PR.

**Comment by darkyne** [comment]
Changed to absolute paths and tested.

**Comment by Drumor** [comment]
Seems ok to me, still needs review from @GuillaumeDerval

**Comment by GuillaumeDerval** [comment]
A remark on this: a "mere" chmod on the directory will not work if the student is root, and this should be noted explicitly somewhere.

Apart from that, i'm ok with this

**Comment by darkyne** [comment]
Yes, obviously if the student has root access on the student_container, this will not prevent him from accessing the files.
I will add a few lines about it in the doc to make it clear.

**Pull request merged**